archetypes/mp-bare/src/main/resources/archetype-resources/src/test/java/MainTest.java.vm

```diff
@@ -0,0 +1,55 @@
+package ${package};
+
+import javax.ws.rs.client.Client;
+import javax.ws.rs.client.ClientBuilder;
+import javax.ws.rs.core.Response;
+
+#if( $mainClass.matches("n|no|false") )
+import io.helidon.microprofile.server.Main;
+#end
+import io.helidon.microprofile.server.Server;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+class MainTest {
+    private static int port;
+
+    @BeforeAll
+    public static void startTheServer() throws Exception {
+#if( $mainClass.matches("n|no|false") )
+        Main.main(new String[0]);
+        port = Main.serverPort();
+#else
+        port = Main.startServer().port();
+#end
+    }
+
+#if( $restResource.matches("y|yes|true") )
+    @Test
+    void testRestResource() {
+        Client client = ClientBuilder.newClient();
+
+        String rsp = client
+                .target(getConnectionString("${restResourcePath}"))
+                .request()
+                .get(String.class);
+        Assertions.assertEquals("It works!", rsp);
+    }
+
+#end
+    @Test
+    void testRoot(){
+        Client client = ClientBuilder.newClient();
+        Response rsp = client
```

Comment:
Nit: `rsp` --> `response`; only a few more characters

Reply:
Roger that. Will change.